### ISSUE TITLE
Ensure-block runs before value is returned 

### ISSUE THREAD

**Issue by judofyr** [jruby/jruby#3388]
```ruby
def wat
  begin
    a = "hello"
    a
  ensure
    a = nil
  end
end

p wat
```

```
$ ruby -v wat.rb
jruby 9.0.1.0 (2.2.2) 2015-09-02 583f336 Java HotSpot(TM) 64-Bit Server VM 24.45-b08 on 1.7.0_45-b18 +jit [darwin-x86_64]
nil
$ ruby -v wat.rb
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
"hello"
```

**Comment by judofyr** [comment]
This causes problems for Erubis + Tilt: rtomayko/tilt#270.

**Comment by enebo** [comment]
Closing as duplicate of https://github.com/jruby/jruby/issues/3356.  This will be released with 9.0.2 in the next week.
